app/models/manageiq/providers/terraform_enterprise/automation_manager.rb

```diff
@@ -8,4 +8,126 @@ def self.ems_type
+
+  def self.params_for_create
+    {
+      :fields => [
+        {
+          :component => 'sub-form',
+          :name      => 'endpoints-subform',
+          :title     => _('Endpoints'),
+          :fields    => [
+            {
+              :component              => 'validate-provider-credentials',
+              :name                   => 'authentications.default.valid',
+              :skipSubmit             => true,
+              :validationDependencies => %w[type zone_id provider_region],
+              :fields                 => [
+                {
+                  :component  => "text-field",
+                  :name       => "endpoints.default.url",
+                  :label      => _("URL"),
+                  :initialValue => "https://app.terraform.io",
+                  :isRequired => true,
+                  :validate   => [{:type => "required"}],
+                },
+                {
+                  :component    => "select",
+                  :id           => "endpoints.default.verify_ssl",
+                  :name         => "endpoints.default.verify_ssl",
+                  :label        => _("SSL verification"),
+                  :dataType     => "integer",
+                  :isRequired   => true,
+                  :validate     => [{:type => "required"}],
+                  :initialValue => OpenSSL::SSL::VERIFY_PEER,
+                  :options      => [
+                    {
+                      :label => _('Do not verify'),
+                      :value => OpenSSL::SSL::VERIFY_NONE,
+                    },
+                    {
+                      :label => _('Verify'),
+                      :value => OpenSSL::SSL::VERIFY_PEER,
+                    },
+                  ]
+                },
+                {
+                  :component  => "password-field",
+                  :id         => "authentications.default.auth_key",
+                  :name       => "authentications.default.auth_key",
+                  :label      => "API Token",
+                  :type       => "password",
+                  :isRequired => true,
+                  :validate   => [{:type => "required"}]
+                },
+              ]
+            }
+          ]
+        }
+      ]
+    }
+  end
+
+  def self.raw_connect(args)
+    default_endpoint       = args.dig("endpoints", "default")
+    default_authentication = args.dig("authentications", "default")
+
+    url, verify_ssl = default_endpoint.values_at("url", "verify_ssl")
+    api_token       = ManageIQ::Password.try_decrypt(default_authentication["auth_key"]) if default_authentication
+    api_token     ||= find(args["id"]).authentication_token("default")
+    url             = File.join(url, "/api/v2")
+
+    Faraday.new(
+      :url     => url,
+      :headers => {
+        "Authorization" => "Bearer #{api_token}",
+        "Content-Type"  => "application/vnd.api+json"
+      },
+      :ssl     => {
+        :verify_mode => verify_ssl
+      }
+    )
```

Comment:
NOTE we should extract this into an API client, so far I haven't found any good definition of the terraform enterprise API other than the website that we could use to auto-generate everything so it'd have to be a manual effort.